src/main/java/com/ecore/roles/client/TeamsClient.java

```diff
@@ -17,21 +19,22 @@
 public class TeamsClient {
 
     private final RestTemplate restTemplate;
+    private final RetryTemplate retryTemplate;
     private final ClientsConfigurationProperties clientsConfigurationProperties;
 
     public ResponseEntity<Team> getTeam(UUID id) {
-        return restTemplate.exchange(
-                clientsConfigurationProperties.getTeamsApiHost() + "/" + id,
-                HttpMethod.GET,
-                null,
-                Team.class);
+        RetryCallback<ResponseEntity<Team>, RuntimeException> callback = context -> restTemplate.getForEntity(
+                clientsConfigurationProperties.getTeamsApiHost() + "/{id}",
+                Team.class,
+                id);
+        return retryTemplate.execute(callback);
     }
 
     public ResponseEntity<List<Team>> getTeams() {
-        return restTemplate.exchange(
+        return retryTemplate.execute(context -> restTemplate.exchange(
```

Comment:
We can improve this client call in order to be a fault tolerant service with spring-retry for 5XX http status response errors